libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts

```diff
@@ -664,6 +707,30 @@ export class SkyDateRangePickerComponent
       });
   }
 
+  #updateRequiredStates(): void {
+    const originalValue = this.outerControlRequired;
+    this.outerControlRequired =
+      this.required ||
+      !!this.#ngControl?.control?.hasValidator(Validators.required);
+
+    if (this.outerControlRequired) {
+      this.#calculatorIdControl?.addValidators(Validators.required);
+    } else {
+      this.#calculatorIdControl?.removeValidators(Validators.required);
+    }
+
+    this.#calculatorIdControl?.updateValueAndValidity();
+    this.#startDateControl?.updateValueAndValidity();
+    this.#endDateControl?.updateValueAndValidity();
+
+    // We need to update the outer control - but only if the required state has changed. Otherwise - we would enter an infinite loop.
+    if (originalValue !== this.outerControlRequired) {
+      this.#ngZone.onStable.pipe(first()).subscribe(() => {
```

Comment:
I added a comment here. It is needed because without it the child control validation may not have finished. I verified this in the playground. Without the `ngZone` the outer control will not show as invalid if the required validator is added until something else retriggers validation. But with this wrapper it will automatically show as invalid when the validator is added.